src/controller.cpp

```diff
@@ -192,6 +194,60 @@ void Controller::activateInternal() {
                    vpnDisabledApps, stateToReason(m_state));
 }
 
+bool Controller::silentSwitchServers() {
+  logger.log() << "Silently switch servers";
+
+  if (m_state != StateOn) {
+    logger.log() << "Cannot silent switch if not on";
+    return false;
+  }
+
+  MozillaVPN* vpn = MozillaVPN::instance();
+  Q_ASSERT(vpn);
+
+  QList<Server> servers = vpn->servers();
+  Q_ASSERT(!servers.isEmpty());
+
+  if (servers.length() <= 1) {
+    logger.log()
+        << "Cannot siltent switch servers because there is only one available";
+    return false;
+  }
+
+  QList<Server>::iterator iterator = servers.begin();
+
+  while (iterator != servers.end()) {
+    if (iterator->publicKey() == vpn->serverPublicKey()) {
+      servers.erase(iterator);
+      break;
+    }
+    ++iterator;
+  }
+
+  Server server = Server::weightChooser(servers);
+  Q_ASSERT(server.initialized());
+
```

Comment:
Maybe check if the current publicKey is != the new one.

Reply:
Since I removed the currently active server previously this should not happen. But I will add an assertion.